packages/devtools/src/devtoolsdriver.ts

```diff
@@ -149,9 +150,13 @@ export default class DevToolsDriver {
         this.activeDialog = dialog
     }
 
-    framenavigatedHandler(frame: Page) {
+    framenavigatedHandler(frame: Frame) {
         this.currentFrameUrl = frame.url()
-        this.elementStore.clear()
+        if (!frame.parentFrame()) {
+            this.elementStore.clear()
+        } else {
+            this.elementStore.clear(frame)
+        }
```

Comment:
How about simplifying it into
```suggestion
        this.elementStore.clear(frame.parentFrame() ? frame : undefined)
```